trip/views.py

```diff
@@ -28,10 +26,21 @@ def create_trip(request):
                 destination_country=dest_country,
             )
 
-            usertrip_instance.trip = trip_instance
-            usertrip_instance.user = request.user
-            usertrip_instance.save()
-            messages.success(request, "Successfully created your trip")
+            usertrip_instance, created = UserTrip.objects.get_or_create(
```

Comment:
I think I understand what you're trying to do here. So we're running get_or_create twice, once to see whether the trip instance is in the database already, and once again to see whether a usertrip instance already exists with these same details?

Reply:
Yes, exactly.

Reply:
V nice solution- can't believe I didn't think of it originally hahaha

Reply:
One thing, though- do you know what parameters get_or_create is checking to determine whether the trip exists in the database or not? If its the primary key, then 

UT A:

PK: 1
Dest: NYC, US
Sdate: 7/20/23
Edate: 7/22/23
User:  request.user
Travel type: companion

and

UT B:

PK: 2
Dest: NYC, US
Sdate: 7/20/23
Edate: 7/22/23
User:  request.user
Travel type: companion

will not be the same even though their attributes are the same. In that case, we might want to define a unique_together attribute in UserTrip's meta class the way one is defined in Trip's meta class. 

It could look like

unique_together = ["trip", "user", "sdate", "edate"] for exact matching
or like unique_together = ["trip", "user", "sdate"] for fuzzy matching, since technically we can't have two trips that start on the same date and a trip to the same dest from 7/20 to 7/23 should be appended to the trip from 7/20 to 7/22 by updating date.

Reply:
So, it's not the PK, which is used to check, it's the other attributes you give, for checking if there exists an entry or not. Like in our case, we are giving start_trip, end_trip, user and trip. If there exists any entry with the exact same values as given, then it won't create any new entry, it would just simply return that particular instance, if while trying to get `Model.DoesNotExist` error was raised, then it would create the new instance and will return that newly created instance.

Reply:
Btw, it's not a good way to give Pk as an attribute to `get_or_create` as this is an unique field, and there won't be ever any instance with the same pk. So, the create will always be triggered.

Reply:
> V nice solution- can't believe I didn't think of it originally hahaha

When I saw you using the it for trip, I thought exactly the same, if you went that far, how come it didn't strike you to do the same here. But anyway, you helped in fixing this one.